### ISSUE TITLE
IsWebSocketRequest_GetProperty_ReturnsExpected failed with "System.Net.HttpListenerException : Address already in use"

### ISSUE THREAD

**Issue by Jiayili1** [dotnet/corefx#20227]
Failed test: System.Net.Tests.HttpListenerRequestTests.IsWebSocketRequest_GetProperty_ReturnsExpected

Mac OSX failure

Detail: https://mc.dot.net/#/product/netcore/200/source/official~2Fcorefx~2Frelease~2F2.0.0~2F/type/test~2Ffunctional~2Fportable~2Fcli~2F/build/20170524.01/workItem/System.Net.HttpListener.Tests/wilogs

~~~
2017-05-23 17:43:46,078: INFO: proc(54): run_and_log_output: Output:    System.Net.Tests.HttpListenerRequestTests.IsWebSocketRequest_GetProperty_ReturnsExpected(webSocketString: \"Connection: \", expected: False) [FAIL]
2017-05-23 17:43:46,078: INFO: proc(54): run_and_log_output: Output:       System.Net.HttpListenerException : Address already in use
2017-05-23 17:43:46,078: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2017-05-23 17:43:46,079: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs(160,0): at System.Net.HttpEndPointManager.GetEPListener(String host, Int32 port, HttpListener listener, Boolean secure, Boolean& alreadyExists)
2017-05-23 17:43:46,079: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs(213,0): at System.Net.HttpEndPointManager.RemovePrefixInternal(String prefix, HttpListener listener)
2017-05-23 17:43:46,079: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs(187,0): at System.Net.HttpEndPointManager.RemoveListener(HttpListener listener)
2017-05-23 17:43:46,079: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/Managed/HttpListener.Managed.cs(161,0): at System.Net.HttpListener.Close(Boolean force)
2017-05-23 17:43:46,079: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/Managed/HttpListener.Managed.cs(147,0): at System.Net.HttpListener.Dispose()
2017-05-23 17:43:46,080: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/src/System/Net/HttpListener.cs(305,0): at System.Net.HttpListener.Close()
2017-05-23 17:43:46,080: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs(613,0): at System.Net.Tests.HttpListenerRequestTests.<GetRequest>d__31.MoveNext()
2017-05-23 17:43:46,080: INFO: proc(54): run_and_log_output: Output:          --- End of stack trace from previous location where exception was thrown ---
2017-05-23 17:43:46,080: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-05-23 17:43:46,080: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:          /Users/buildagent/agent/_work/31/s/corefx/src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs(205,0): at System.Net.Tests.HttpListenerRequestTests.<IsWebSocketRequest_GetProperty_ReturnsExpected>d__12.MoveNext()
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:          --- End of stack trace from previous location where exception was thrown ---
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:          --- End of stack trace from previous location where exception was thrown ---
2017-05-23 17:43:46,081: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-05-23 17:43:46,082: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-05-23 17:43:46,082: INFO: proc(54): run_and_log_output: Output:          --- End of stack trace from previous location where exception was thrown ---
2017-05-23 17:43:46,082: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-05-23 17:43:46,082: INFO: proc(54): run_and_log_output: Output:             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
~~~

**Comment by karelz** [comment]
cc @hughbe @stephentoub - one time Mac OSX failure, moving to 2.1.
If you know what's wrong from top of your head, we could fix it in 2.0, because these are new tests ...

**Comment by karelz** [comment]
Only one more failure  of the test from official Helix runs:
* Day: 8/26
* Build: 20170831.01
* Error message: System.Net.Internals.SocketExceptionFactory+ExtendedSocketException : Connection refused 127.0.0.1:1444

Closing.
Please reopen if it happens again.
